kobo/apps/audit_log/models.py

```diff
@@ -94,12 +95,54 @@ def save(
             update_fields=update_fields,
         )
 
+
+class AccessLogManager(models.Manager):
+    def get_queryset(self):
+        return super().get_queryset().filter(log_type=AuditType.ACCESS)
+
+    def create(self, **kwargs):
+        # remove any attempt to set fields that should always be the same on an access log
+        app_label = kwargs.pop('app_label', None)
+        if app_label is not None:
+            logging.warning(
+                f'Ignoring attempt to set {app_label=} on access log'
+            )
+        model_name = kwargs.pop('model_name', None)
+        if model_name is not None:
+            logging.warning(
+                f'Ignoring attempt to set {model_name=} on access log'
+            )
+        action = kwargs.pop('action', None)
+        if action is not None:
+            logging.warning(f'Ignoring attempt to set {action=} on access log')
+        log_type = kwargs.pop('log_type', None)
+        if log_type is not None:
+            logging.warning(
+                f'Ignoring attempt to set {log_type=} on access log'
+            )
+        user = kwargs.pop('user')
+        return super().create(
+            # set the fields that are always the same for access logs, pass along the rest to the original constructor
+            app_label=ACCESS_LOG_KOBO_AUTH_APP_LABEL,
+            model_name=User.__qualname__,
```

Comment:
`User.__qualname__` returns `User`, but model_name should be in lowercase. 
At the first glance, the idea was to be consistent with `ContentType` model and `Meta` class attributes. 

```python
from django.contrib.contenttypes.models import ContentType

c = ContentType.objects.get_for_model(User)
c.app_label  # kobo_auth 
c.model  # user <-- lowercase

User._meta.app_label  # kobo_auth 
User._meta.model_name  # user
```

Thinking about that, do we need really need a constant for `ACCESS_LOG_KOBO_AUTH_APP_LABEL = 'kobo_auth'`  since we can get it directly with the `Meta` class attributes of User class ?

Reply:
Should I add a migration for old audit logs to use lowercase?